modules/distr-tracing-tempo-install-cli.adoc

```diff
@@ -0,0 +1,176 @@
+<object_storage_secret>
+EOF
+----
+====
++
+--
+include::snippets/distr-tracing-tempo-required-secret-parameters.adoc[]
+--
++
+--
+include::snippets/distr-tracing-tempo-secret-example.adoc[]
+--
+
+. Use `oc create namespace` or `oc apply` to create a namespace for the *TempoStack* instance that you will create in the next step.
++
+[TIP]
+====
+[source,terminal]
+----
+$ oc create namespace <namespace_value_of_your_choice>
+----
+
+[source,terminal]
+----
+$ oc apply -f - << EOF
+apiVersion: project.openshift.io/v1
+kind: Project
+metadata:
+  labels:
+    kubernetes.io/metadata.name: <namespace_value_of_your_choice>
+    openshift.io/cluster-monitoring: "true"
+  name: <namespace_value_of_your_choice>
+EOF
+----
+====
+
+. Create a *TempoStack* instance in the namespace that you created for the *TempoStack* instance in the previous step.
++
+NOTE: You can create multiple *TempoStack* instances in separate namespaces on the same cluster.
++
+.. Customize the `TempoStack` custom resource (CR):
++
+[source,yaml]
+----
+apiVersion: tempo.grafana.com/v1alpha1
+kind: TempoStack
+metadata:
+  name: sample
+  namespace: <namespace_from_previous_step>
+spec:
+  storageSize: 1Gi
+  storage:
+      secret:
+          name: <secret-name> <1>
+          type: <secret-provider> <2>
+  template:
+    queryFrontend:
+      jaegerQuery:
+        enabled: true
+        ingress:
+          route:
+            termination: edge
+          type: route
+----
+<1> The value of `name` in the `metadata` of the Secret.
+<2> The accepted values are `azure` for Azure Blob Storage, `gcs` for Google Cloud Storage, and `s3` for Amazon S3 or MinIO or Red Hat OpenShift Data Foundation.
++
+.TempoStack CR for AWS S3 and MinIO storage
+====
+[source,yaml]
+----
+apiVersion: tempo.grafana.com/v1alpha1
+kind: TempoStack
+metadata:
+  name: simplest
+  namespace: tempostacknamespace
+spec:
+  storageSize: 1Gi
+  storage:
+    secret:
+      name: minio-test
+      type: s3
+  resources:
+    total:
+      limits:
+        memory: 2Gi
+        cpu: 2000m
+  template:
+    queryFrontend:
+      jaegerQuery:
+        enabled: true
+        ingress:
+          route:
+            termination: edge
+          type: route
+----
+The stack deployed in this example is configured to receive Jaeger Thrift over HTTP and OpenTelemetry Protocol (OTLP), which permits visualizing the data with the Jaeger UI.
+====
+
+.. Apply the customized CR.
++
+[source,terminal]
+----
+$ oc apply -f - << EOF
+<TempoStack_custom_resource>
+EOF
+----
+
+link:https://github.com/openshift/openshift-docs/pull/62410/files#r1284356025[Waiting for SME input for the following:]
+
+* Command(s) to verify successful deployment of the TempoStack instance
++
+For example:
+** A command that prints output to verify that pods are created on the namespace.
+** A command that enables the CLI user to watch for the stack to stabilize.
+
+* If applicable, a command to open any UI console
+* Any other commands?
```

Comment:
@rubenvp8510 and @IshwarKanse, when you can find the time, could you come up with suggested changes for these lines here?

Reply:
For checking the TempoStack status, we can use these commands.
1. Make sure all the component status is running and the condition in status is in Ready state.
```
$ oc get tempostacks.tempo.grafana.com simplest -o yaml
apiVersion: tempo.grafana.com/v1alpha1
kind: TempoStack
metadata:
  creationTimestamp: "2023-08-07T07:46:48Z"
  generation: 1
  labels:
    app.kubernetes.io/managed-by: tempo-operator
    tempo.grafana.com/distribution: openshift
  name: simplest
  namespace: tracing-system
  resourceVersion: "60486"
  uid: 4f68828e-8233-4d50-9885-96b663b030f7
spec:
  images:
.............REDACTED ......
  limits:
    global:
      ingestion: {}
      query:
        maxSearchDuration: 0s
  managementState: Managed
  observability:
    metrics: {}
    tracing:
      jaeger_agent_endpoint: localhost:6831
  replicationFactor: 1
  resources:
    total:
      limits:
        cpu: "2"
        memory: 2Gi
  retention:
    global:
      traces: 48h0m0s
  search:
    defaultResultLimit: 20
    maxDuration: 0s
  serviceAccount: tempo-simplest
  storage:
    secret:
      name: minio-secret
      type: s3
  storageSize: 20Gi
  template:
    compactor: {}
    distributor:
      replicas: 1
    gateway:
      component: {}
      enabled: false
      ingress:
        route: {}
    ingester:
      replicas: 1
    querier: {}
    queryFrontend:
      component: {}
      jaegerQuery:
        enabled: true
        ingress:
          route:
            termination: edge
          type: route
status:
  components:
    compactor:
      Running:
      - tempo-simplest-compactor-77cc4b9c9f-7grk4
    distributor:
      Running:
      - tempo-simplest-distributor-6759b88d99-wdqfn
    ingester:
      Running:
      - tempo-simplest-ingester-0
    querier:
      Running:
      - tempo-simplest-querier-74d77fc548-f4nxd
    queryFrontend:
      Running:
      - tempo-simplest-query-frontend-54b7c465b5-fdrxs
  conditions:
  - lastTransitionTime: "2023-08-07T07:47:12Z"
    message: Some TempoStack components are pending on dependencies
    reason: PendingComponents
    status: "False"
    type: Pending
  - lastTransitionTime: "2023-08-07T07:47:12Z"
    message: All components are operational
    reason: Ready
    status: "True"
    type: Ready
  operatorVersion: 0.3.0
  tempoVersion: 2.1.1
```
2. Make sure all the component pods are running.
```
$ oc get pods
NAME                                             READY   STATUS    RESTARTS   AGE
tempo-simplest-compactor-77cc4b9c9f-7grk4        1/1     Running   0          3m1s
tempo-simplest-distributor-6759b88d99-wdqfn      1/1     Running   0          3m1s
tempo-simplest-ingester-0                        1/1     Running   0          3m1s
tempo-simplest-querier-74d77fc548-f4nxd          1/1     Running   0          3m1s
tempo-simplest-query-frontend-54b7c465b5-fdrxs   2/2     Running   0          3m1s
```
To get the route for the TempoStack query UI, run the following command.
```
$ oc get route -n tempostacknamespace
NAME                            HOST/PORT                                                                                     PATH   SERVICES                        PORT        TERMINATION   WILDCARD
tempo-simplest-query-frontend   tempo-simplest-query-frontend-tempostacknamespace.apps.mytest-55.qe.gcp.mycluster.test.com          tempo-simplest-query-frontend   jaeger-ui   edge          None

```